lib/client.js

```diff
@@ -179,6 +179,12 @@ var warmupLimit = 32;
  *   determine if an statement can be retried in case of request error or write timeout.
  * </p>
  * <p>Default: <code>false</code>.</p>
+ * @property {String} [keyspace] Specifies the keyspace for the query. Used for routing within the driver, this
```

Comment:
This does seem useful for cases where you are executing a non-prepared query where the user provides the routing key explicitly and the keyspace cannot be discerned other than what is set on the connection (which may not be the same).   I think all drivers should maybe have an option for this (i.e. the java driver doesn't have this option as far as I know).

One thing that this could be conflated with eventually is [CASSANDRA-10145](https://issues.apache.org/jira/browse/CASSANDRA-10145) where at the protocol level we can set the keyspace for a statement.  That will be really nice to have, but one thing I worry about is if we introduce this query option now and then eventually overload it to also set the keyspace on the executed query at the protocol level, although on the other hand it wouldn't cause any issues because the user's intent after all is to declare that this is the keyspace the query uses.